Runtime/AvatarCreator/UI/Elements/SelectionElement.cs

```diff
@@ -29,6 +33,23 @@ public ButtonElement CreateButton(string id)
             return button;
         }
 
+        public void CreateButtons<T>(T[] assets, Action<ButtonElement, T> onButtonCreated = default) where T : IAssetData
```

Comment:
don't you think that this make it look more complicated (and potentially harder to write simple documentation for) with the use of generics and function passing? 
I understand that you wanted to remove some code duplication though.

Reply:
I like generics in general, but this could be overengineering, though I don't know any other way to remove duplication right now.

Reply:
Personally, I don't really see it as an issue, because it is a generic in a lower level part of the element? The core documentation for partners would be that they could easily extend this class with their own functions to create their own selector, and they get a helpful callback to edit the button style if they want to.

IMO this is easier to understand and extend than it would be the other way around since there's less uniformity otherwise.

In my mind this strips away the code that the partners don't **need** care about, but of course they can if they want to.